### ISSUE TITLE
Kernel bug: "unregister_netdevice: waiting for lo to become free. Usage count = 1"

### ISSUE THREAD

**Issue by lantz** [mininet/mininet#164]
On certain kernels (notably Ubuntu 12.04.2 kernel?) the `unshare()` system call can hang. This breaks `mnexec` and Mininet!! :(

https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1197078

**Comment by lantz** [comment]
It's back in Ubuntu 16.04, unfortunately.

https://bugzilla.kernel.org/show_bug.cgi?id=81211

**Comment by RRAlex** [comment]
Cross posted here before seeing this:
https://github.com/docker/docker/issues/5618#issuecomment-240120471

But indeed, it seems to have cropped back recently in 16.04, I never had this before and now 2 VMs and my laptop required a hard reboot within the last few days. (kernel 4.4.0-3[456] probably)
Most, but not all, were on 1.12. So probably a kernel bug indeed...

**Comment by lantz** [comment]
Yeah it's very annoying. Ideally we can identify a kernel that works on 16.04 and doesn't exhibit this bug.
